src/libcore/char.rs

```diff
@@ -299,7 +299,17 @@ impl CharExt for char {
 
     #[inline]
     fn escape_unicode(self) -> EscapeUnicode {
-        EscapeUnicode { c: self, state: EscapeUnicodeState::Backslash }
+        let c = self as u32;
+        // or-ing 1 ensures that for c==0 the code computes that one
+        // digit should be printed and (which is the same) avoids the
+        // (31 - 32) underflow
+        let msb = 31 - (c | 1).leading_zeros();
+        let msdigit = msb / 4;
```

Comment:
I am afraid that could be misleading, because it is not the number of (hex)digits, but the index of the most significant one.
I hoped that the parallel with MSB would guide the reader and I thought that that presenting it as number of `hex_digits` would make them believe that an `hex_digits = 0` would not be acceptable, while it is the correct offset for any `c` between `0` and `0xf`.